### PR TITLE
avoid deadlock by skipping sampling in libc, libgcc and pthread

### PR DESCRIPTION
**Pull request by YangKeao**
Signed-off-by: YangKeao <yangkeao@chunibyo.icu>

### CODE REVIEW

**Comment by YangKeao** [comment]
@BusyJay PTAL

File: src/profiler.rs [hunk 1]
```diff
@@ -23,6 +26,92 @@ pub struct Profiler {
     sample_counter: i32,
 
     running: bool,
+
+    #[cfg(all(any(target_arch = "x86_64", target_arch = "aarch64")))]
+    blacklist_segments: Vec<(usize, usize)>,
```
Comment (BusyJay) [inline src/profiler.rs:31]: How about `blocklist_segments`?

**Review by BusyJay — COMMENTED** [review]
Rest LGTM

File: src/profiler.rs (hunk 1 above)
Comment (YangKeao) [reply src/profiler.rs]: Good Idea :+1:

**Review by BusyJay — APPROVED** [review]

**Comment by sticnarf** [comment]
A simpler solution might be hacking `dl_iterate_phdr`. Set a thread local flag when `dl_iterate_phdr` is called, then we can skip the sample by checking the thread local flag.

**Pull request merged**